packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/ItemCharts/StockDistribution.tsx

```diff
@@ -112,18 +114,56 @@ const MonthlyBar = ({
   );
 };
 
+const CalculationError = ({
+  isAmcZero,
+  isSohAndQtyZero,
+}: {
+  isAmcZero?: boolean;
+  isSohAndQtyZero?: boolean;
+}) => {
+  const t = useTranslation('replenishment');
+  const detail = isAmcZero
+    ? `: ${t('error.amc-is-zero')}`
+    : isSohAndQtyZero
+    ? `: ${t('error.soh-and-suggested-quantity-are-zero')}`
+    : '';
+  const message = `${t('error.unable-to-calculate')}${detail}`;
+
+  return (
+    <Box
+      sx={{
+        paddingLeft: 4,
+        paddingRight: 4,
+        paddingTop: 4,
+        paddingBottom: 2,
+      }}
+    >
+      <Typography variant="body1" fontWeight={700} style={{ marginBottom: 10 }}>
+        {t('heading.stock-distribution')}
+      </Typography>
+      <Box display="flex" padding={1} gap={1}>
+        <AlertIcon color="primary" fontSize="small" />
+        <Typography variant="body1" fontSize={12} sx={{ color: 'error.main' }}>
+          {message}
+        </Typography>
+      </Box>
+    </Box>
+  );
+};
+
 export const StockDistribution: React.FC<StockDistributionProps> = ({
   availableStockOnHand = 0,
   averageMonthlyConsumption = 0,
   suggestedQuantity = 0,
 }) => {
-  if (averageMonthlyConsumption === 0) return null;
+  if (averageMonthlyConsumption === 0) return <CalculationError isAmcZero />;
 
   const { maxMonthsOfStock } = useRequestFields('maxMonthsOfStock');
   const targetQuantity = maxMonthsOfStock * averageMonthlyConsumption;
   const t = useTranslation('replenishment');
 
-  if (suggestedQuantity + availableStockOnHand === 0) return null;
+  if (suggestedQuantity + availableStockOnHand === 0)
+    return <CalculationError isSohAndQtyZero />;
```

Comment:
Maybe I'm not quite understanding this -- but the expression doesn't quite appear to match what the UI message says: "No Stock on Hand or Suggested Quantity values".

Wouldn't the condition be `if (suggestedQuantity === 0 || availableStockOnHand === 0)` (or maybe `  if (!suggestedQuantity || !availableStockOnHand)` to allow for undefined/null as well as 0)

Reply:
the values are defaulting to 0

```
  availableStockOnHand = 0,
  averageMonthlyConsumption = 0,
  suggestedQuantity = 0,
```
so can't be undefined.
And `suggestedQuantity + availableStockOnHand === 0` is equivalent to both of the vars being 0 which is what I'm testing for. was really just using shorthand, but will stretch it out to

 `if (suggestedQuantity === 0 && availableStockOnHand === 0)`

to make is clearer.

Reply:
Oh right, it's *BOTH*, not *either*. I think the "or" in the message string confused me. The trouble with English "or" meaning something slightly different to logical "OR" 😄

Reply:
I'm really not sure about the wording on that error message! that just confirms it 😀